packages/store/operators/src/patch.ts

```diff
@@ -9,11 +10,12 @@ type PatchValues<T> = {
 
 type PatchOperator<T> = <U extends PatchValues<T>>(existing: Readonly<U>) => U;
 
-export function patch<T>(patchObject: PatchSpec<T>): PatchOperator<T> {
+export function patch<T>(patchObject: PatchSpec<NoInfer<T>>): PatchOperator<T> {
   return function patchStateOperator<U extends PatchValues<T>>(existing: Readonly<U>): U {
     let clone = null;
-    for (const k in patchObject) {
-      const newValue = patchObject[k];
+    const thePatchObject = <PatchSpec<T>>patchObject;
+    for (const k in thePatchObject) {
+      const newValue = thePatchObject[k];
```

Comment:
although `T` should not be inferred from `patchObject`, we still expect that the actual shape of `patchObject` will be `PatchSpec<T>`